module/zstd/Makefile.in

```diff
@@ -0,0 +1,38 @@
+src = @abs_top_srcdir@/module/zstd
+obj = @abs_builddir@
+
+MODULE := zzstd
+
+obj-$(CONFIG_ZFS) := $(MODULE).o
+
+asflags-y += $(ZFS_MODULE_CFLAGS)
+ccflags-y += $(ZFS_MODULE_CFLAGS) $(ZFS_MODULE_CPPFLAGS)
+
+# Suppress unused but set variable warnings often due to ASSERTs
+ccflags-y += $(NO_UNUSED_BUT_SET_VARIABLE)
+ccflags-y += -Wframe-larger-than=20480 -O3
```

Comment:
Its important to add a comment here about the `-Wframe-larger-than=20480`
Somthing like:
`-Wframe-larger-than=20480 needed, due to requirements of the stock zstd library`

To prevent confusion like here:
https://github.com/zfsonlinux/zfs/pull/8044#discussion_r226809224

Reply:
I wonder if we should include $(FRAME_LARGER_THAN) here to make this overrideable like in lib/libzpool/Makefile.am (even here it's wrong IMHO)

Reply:
And we have another "problem": "behlendorf: Please drop, this is already set to 4096 for kernel builds (which is already flirting with way to large)."

Reply:
This discission has been had before, sadly enough ZSTD requires it on default.

Your other include I dont get....

Reply:
Well, see `lib/libzpool/Makefile.am` - here the frame size gets used from `configure`. We override that unconditionally, even if someone for some reason specified a greater framesize (lower would not work ofc).

Reply:
Now I get it, sorry was too much focussed on those damn reviews...

Yeah I think you should make it overrideable, considering $(FRAME_LARGER_THAN) is used quite standard...

Reply:
ok, not needed for libzpool, only for the kernel build

Reply:
forget it... I was wrong.. and this is even the wrong comment \o/

Reply:
hmm, making this overridable is a bit difficult... FRAME_LARGER_THAN gets set to `-Wframe-larger-than=4096` by default. We would need to compare the integer value and force it to 20480 if <20480.... something like this maybe (pseudocode! ;) ):
if int(FRAME_LARGER_THAN) == 4096 || FRAME_LARGER_THAN == ""
  FRAME_LARGER_THAN = -Wframe-larger-than=20480
else
  keep FRAME_LARGER_THAN as we assume, the user has overridden this

Reply:
Maybe that would get a little too dirty....
And the frame_larger_than here is only for our module and is already very-very big.